src/ai/defaultai.cc

```diff
@@ -1174,6 +1178,14 @@ void DefaultAI::late_initialization() {
 				ports_vicinity.insert(hash);
 			}
 		} while (mr.advance(map));
+		Widelands::HollowArea<> ha(Widelands::Area<>(map.get_fcoords(c), 8), 3);
+		Widelands::MapHollowRegion<> mhr(map, ha);
+		do {
+			const uint32_t hash = mhr.location().hash();
+			if (!ports_shipyard_region.count(hash)) {
+				ports_shipyard_region.insert(hash);
+			}
```

Comment:
Calling `insert()` on a set or unordered_set does nothing if the key is already present
```suggestion
			ports_shipyard_region.insert(hash);
```

Reply:
didn't know that. In fact I tend to implement things by "copy + paste + adjust" as this is the same method as a couple of lines above I think I should go over the complete code to clean these things up.  
do I understand this right that the suggested code might be more performant as it gets rid of an if check.